src/components/App.js

```diff
@@ -9,16 +8,39 @@ class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      result: '0',
+      total: null,
+      next: null,
+      operation: null,
     };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    const { total, next, operation } = this.state;
+    const copyObj = { total, next, operation };
+    const result = calculate(copyObj, buttonName);
+    this.setState({
+      total: result.total,
+      next: result.next,
+      operation: result.operation,
+    });
   }
 
   render() {
-    const { result } = this.state;
+    const { total, next } = this.state;
+    const res = next || total;
+    if (res) {
+      return (
+        <div className="app-wrapper">
+          <Display result={res} />
+          <ButtonPanel clickHandler={(buttonName) => this.handleClick(buttonName)} />
+        </div>
+      );
+    }
     return (
       <div className="app-wrapper">
-        <Display result={result} />
-        <ButtonPanel />
+        <Display />
+        <ButtonPanel clickHandler={(buttonName) => this.handleClick(buttonName)} />
       </div>
     );
   }
```

Comment:
[Optional]

I think these two `return` is unnecessary. When `res` is `undefined` the `defaultProps` is used. returning only the first case will do.
Let's delete the second case and return `res` as `result` props